src/main/java/com/puppycrawl/tools/checkstyle/site/ViolationMessagesMacro.java

```diff
@@ -0,0 +1,126 @@
+///////////////////////////////////////////////////////////////////////////////////////////////
+// checkstyle: Checks Java source code and other text files for adherence to a set of rules.
+// Copyright (C) 2001-2023 the original author or authors.
+//
+// This library is free software; you can redistribute it and/or
+// modify it under the terms of the GNU Lesser General Public
+// License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+//
+// This library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+// Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public
+// License along with this library; if not, write to the Free Software
+// Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
+///////////////////////////////////////////////////////////////////////////////////////////////
+
+package com.puppycrawl.tools.checkstyle.site;
+
+import java.util.Collections;
+import java.util.List;
+
+import org.apache.maven.doxia.macro.AbstractMacro;
+import org.apache.maven.doxia.macro.Macro;
+import org.apache.maven.doxia.macro.MacroExecutionException;
+import org.apache.maven.doxia.macro.MacroRequest;
+import org.apache.maven.doxia.module.xdoc.XdocSink;
+import org.apache.maven.doxia.sink.Sink;
+import org.codehaus.plexus.component.annotations.Component;
+
+/**
+ * A macro that inserts a list of the violation messages.
+ */
+@Component(role = Macro.class, hint = "violation-messages")
+public class ViolationMessagesMacro extends AbstractMacro {
+    @Override
+    public void execute(Sink sink, MacroRequest request) throws MacroExecutionException {
+        // until https://github.com/checkstyle/checkstyle/issues/13426
+        if (!(sink instanceof XdocSink)) {
+            throw new MacroExecutionException("Expected Sink to be an XdocSink.");
+        }
+        final String checkName = (String) request.getParameter("checkName");
+        final Object instance = SiteUtil.getModuleInstance(checkName);
+        final Class<?> clss = instance.getClass();
+        final List<String> messageKeys = SiteUtil.getCheckMessageKeys(clss, instance);
+        createListOfMessages((XdocSink) sink, clss, messageKeys);
```

Comment:
So what is the plan in terms of disabling javadoc. Are we not going to have these type of generation in meta and have our macros get the data from meta? To me this makes the most sense, instead of meta having another custom implementation.

Reply:
This implementation will substitute  getting data from metadata.
For now validation javadoc vs xdoc help to test this implementation until it gets its maturity.
Such "another implementation" will me main implementation and source of truth soon.